ecs-cli/modules/cli/compose/entity/entity_helper.go

```diff
@@ -188,6 +187,10 @@ func getContainersForTasks(entity ProjectEntity, ecsTasks []*ecs.Task) ([]compos
 		var ec2IPAddress string
 		if ec2ID != "" && ec2Instances[ec2ID] != nil {
 			ec2IPAddress = aws.StringValue(ec2Instances[ec2ID].PublicIpAddress)
+			if ec2IPAddress == "" {
+				// If Public IP is empty, use private IP
```

Comment:
Nit: Personally I feel like the code is sufficiently self-commenting, but fine with leaving this in.

Reply:
This is a personally preference thing, but I like leaving small comments like this, because sure you could easily read the code and see what it does, but this makes it so that the code can be understood a lot faster.

Reply:
I agree with @SoManyHs, the best practice is to let the code speak itself, and only add comment if the code is not as obvious.